src/shared/tpm2-util.c

```diff
@@ -1866,6 +1866,150 @@ int tpm2_make_luks2_json(
         return keyslot;
 }
 
+int tpm2_parse_luks2_json(
+                JsonVariant *v,
+                int *ret_keyslot,
+                uint32_t *ret_hash_pcr_mask,
+                uint16_t *ret_pcr_bank,
+                void **ret_pubkey,
+                size_t *ret_pubkey_size,
+                uint32_t *ret_pubkey_pcr_mask,
+                uint16_t *ret_primary_alg,
+                void **ret_blob,
+                size_t *ret_blob_size,
+                void **ret_policy_hash,
+                size_t *ret_policy_hash_size,
+                TPM2Flags *ret_flags) {
+
+        _cleanup_free_ void *blob = NULL, *policy_hash = NULL, *pubkey = NULL;
+        size_t blob_size = 0, policy_hash_size = 0, pubkey_size = 0;
+        uint32_t hash_pcr_mask = 0, pubkey_pcr_mask = 0;
+        uint16_t primary_alg = TPM2_ALG_ECC; /* ECC was the only supported algorithm in systemd < 250, use that as implied default, for compatibility */
+        uint16_t pcr_bank = UINT16_MAX; /* default: pick automatically */
```

Comment:
shouldn't this be the sha256 bank, given the comment below about being a new option post v250 and defaulting to sha256?

Reply:
if the bank is not specified we communicate that as UINT16_MAX to the caller. Which tells the caller "figure it out yourself". The caller then checks which banks the TPM and the firmware support, and picks SHA256 if both support it, and SHA1 otherwise.

What's important is that during enrollment and during unlocking we use the same bank. If we auto-detect the bank we might come to different results though (for example: when volume was enrolled fimrware could only do SHA1, but a firmware update was done, which added SHA256 support and the TPM always could do it). Hence, in current versions we generally store away the bank we picked to avoid any discrepancies there. But then we need a way to communicate the "please audo-detect" mode which is what UINT16_MAX does here.

Reply:
got it, thanks